test/NuGet.Core.Tests/NuGet.Common.Test/RuntimeEnvironmentHelperTests.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace NuGet.Common.Test
+{
+    public class RuntimeEnvironmentHelperTests
+    {
+        [Fact]
+        public void RuntimeEnvironmentHelper_MutuallyExclusiveRuntimes()
+        {
+            // It only should return true in either Windows or MacOS/Linux
+            Assert.True(RuntimeEnvironmentHelper.IsWindows ^ (RuntimeEnvironmentHelper.IsMacOSX | RuntimeEnvironmentHelper.IsLinux));
```

Comment:
I don't understand this test at all. you can't be on windows and mac or on windows and linux at the same time. Even Windows Subsystem for Linux should return false for IsWindows and true for IsLinux.

Reply:
The basic idea is to have only one flag depending on the platform you're running. Good catch on WSL!